packages/server/src/service/util.ts

```diff
@@ -8,6 +8,46 @@ interface Options{
 }
 type AsyncFunc<A extends unknown[], R> = (...args: A) => Promise<R>
 
+/**
+ * Asynchronously reads the entire contents of a file.
+ *
+ * @param path A path to a file. If a URL is provided, it must use the
+ * file: protocol. If a file descriptor is provided, the underlying file
+ * will not be closed automatically.
+ *
+ * @param options An object that may contain an optional flag. If a flag
+ * is not provided, it defaults to 'r'.
+ */
+export const safeReadFile = async (
+  path: fs.PathLike | number,
+  options?: { encoding?: null | undefined, flag?: string | undefined },
+) => new Promise<Uint8Array>((resolve, reject) => {
```

Comment:
Make this return `Promise<Buffer>` directly via `resolve(Buffer.from(data.buffer))`.  I think we need the `Buffer.from` to ensure it's copied (although I might be wrong).